api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3455,6 +3455,34 @@ public void saveOrder_shouldRevisePreviousNonCodedOrderIfItIsAlreadyExisting() {
 		assertTrue(saveOrder.getAsNeeded());
 		assertNotNull(orderService.getOrder(saveOrder.getOrderId()));
 	}
+	
+	/**
+	 * @see org.openmrs.api.OrderService#saveOrder(Order, OrderContext)
+	 */
+	@Test
+	public void saveOrder_shouldSetNonCodedDrugOrderConcept() {
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-nonCodedDrugs.xml");
+
```

Comment:
Is this testing the changes in OrderServiceImpl or the ones in DrugOrderValidator?

Reply:
It's testing the changes in OrderServiceImpl

Reply:
What do you think of also testing the changes in the validator?